framework/Models.py

```diff
@@ -297,6 +299,21 @@ def _handleInput(self, paramInput):
       self.raiseADebug(" Failed in Node: "+str(xmlNode),verbostiy='silent')
       self.raiseAnError(IOError,'missed subType for the model '+self.name)
 
+  @abc.abstractmethod
+  def evaluateSample(self, myInput, samplerType, kwargs):
+    """
+        This will evaluate an individual sample on this model. Note, parameters
+        are needed by createNewInput and thus descriptions are copied from there.
+        @ In, myInput, list, the inputs (list) to start from to generate the new one
+        @ In, samplerType, string, is the type of sampler that is calling to generate a new input
+        @ In, **kwargs, dict,  is a dictionary that contains the information coming from the sampler,
+           a mandatory key is the sampledVars'that contains a dictionary {'name variable':value}
+        @ Out, returnValue, dict, This holds the output information of the evaluated sample.
+    """
+    # inRun = self.createNewInput(myInput, samplerType, **kwargs)
+    # output = self.run(inRun)
+    pass
```

Comment:
Right, the abstract method has no return value because you cannot call an abstract method, but everyone who implements it should have a returnValue that is a dict that holds the output information of the evaluated sample. If we don't specify this here then any new subclasses that get implemented run the risk of violating this contract, or people who use one or more subclasses will not know what to expect without looking at one or more subclasses. 

I believe this is generic information that should be available at the most generic level, but if you have a counter example I would be open to discussion.